src/argilla/server/contexts/datasets.py

```diff
@@ -821,6 +826,16 @@ async def _preload_record_relationships_before_index(db: "AsyncSession", record:
     )
 
 
+async def preload_records_relationships_before_validate(db: "AsyncSession", records: List[Record]) -> None:
+    await db.execute(
+        select(Record)
+        .filter(Record.id.in_([record.id for record in records]))
+        .options(
+            selectinload(Record.dataset).selectinload(Dataset.questions),
+        )
+    )
```

Comment:
I think this could be faster getting the unique datasets ids from the list of provided records, and the selecting the datasets instead of the records.

Reply:
We already reduce the time consumption of the endpoint x2. I would say not to apply more optimization on fetching data. It would be interesting to improve the time consumption on the update part. But this can be tackled in future work.

<img width="914" alt="Screenshot 2024-01-08 at 11 45 34" src="https://github.com/argilla-io/argilla/assets/2518789/8f1aef6a-2c14-4015-aada-9aa65e3e4fda">